core/domain/user_services.py

```diff
@@ -495,6 +495,96 @@ def get_user_settings_by_auth_id(
         return None
 
 
+def get_all_user_groups() -> List[Dict[str, Union[str, List[str]]]]:
+    """Return the list of user groups dictionary data having key as user-group
+    name and value as list of users associated to the user-group.
+
+    Returns:
+        user_group_models_dict: List[Dict[str, Union[str, List[str]]]]. List of
```

Comment:
You should be returning domain objects internally, not dicts.

Also I think user_group_models_dict is not the right name because the dict doesn't contain models in it. Differentiate clearly between storage models (which should not be used except at the interface to the datastore) and domain objects (which are plain Python objects that can be used throughout the code).

Reply:
Make sense, I have updated this.